runtime/queries/metricsview_data_types.go

```diff
@@ -0,0 +1,171 @@
+package queries
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"io"
+	"strings"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+type MetricsViewDataTypes struct {
+	MetricsViewName    string                               `json:"metrics_view_name,omitempty"`
+	MetricsView        *runtimev1.MetricsViewSpec           `json:"-"`
+	ResolvedMVSecurity *runtime.ResolvedMetricsViewSecurity `json:"security"`
```

Comment:
In the alerts PR, I'm making some refactors to stop passing full metrics view specs in here. Would recommend doing the same here (can wait until the other PR is merged so the correct funcs are available in this package). See [this diff](https://github.com/rilldata/rill/pull/3783/files#diff-4a01247b922ef6106d44f5553dc3fba7130fadb020de3ff8260e29b71f9c4e8e) for `runtime/queries/metricsview_aggregation.go` for an example.